jax/dtypes.py

```diff
@@ -120,7 +122,8 @@ def coerce_to_array(x):
 def finfo(dtype):
   # Since NumPy doesn't consider bfloat16 a floating-point type, we have to
   # provide an alternative implementation of finfo that does so.
-  if np.result_type(dtype) == _bfloat16_dtype:
+  if ((isinstance(dtype, str) and dtype == "bfloat16") or
+      np.result_type(dtype) == _bfloat16_dtype):
```

Comment:
Are these two different kinds of bfloat16 types?

Reply:
This update lets the function handle ``"bfloat16"`` as a string. ``np.result_type`` accepts string inputs, but doesn't know about bfloat16.